src/vs/vscode.d.ts

```diff
@@ -6538,11 +6538,28 @@ declare module 'vscode' {
 		getParent?(element: T): ProviderResult<T>;
 	}
 
+	/**
+	 * Label describing the [Tree item](#TreeItem)
+	 */
+	export interface TreeItemLabel {
```

Comment:
Having this as a separate type means, this will only work when the label is a string, right. Or putting it the other way, highlights will not work when the label is an `Uri`, right?

Reply:
Yes. You can only provide highlights to the labels you contribute. Otherwise it is not correct to allow providing highlights to computed labels.

Reply:
Yes, you can only have highlights for labels you contribute, otherwise it is not correct to allow them for computed labels.